components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java

```
@@ -444,4 +466,41 @@ public static void deleteTenantUMData(int tenantId) throws Exception {
 
         TenantUMDataDeletionUtil.deleteTenantUMData(tenantId, dataSource.getConnection());
     }
+
+    /**
+     * Broadcast TenantDeleteClusterMessage to all worker nodes
+     *
+     * @param tenantId
+     * @throws Exception
+     */
+    public static void deleteWorkernodesTenant(int tenantId) throws Exception {
+        TenantDeleteClusterMessage clustermessage = new TenantDeleteClusterMessage(
+                tenantId);
+        ConfigurationContext configContext = TenantMgtServiceComponent.getConfigurationContext();
+        ClusteringAgent agent = configContext.getAxisConfiguration()
+                .getClusteringAgent();
+        try {
+            agent.sendMessage(clustermessage, true);
+        } catch (ClusteringFault e) {
+            log.error("Error occurred while broadcasting TenantDeleteClusterMessage : " + e.getMessage());
```

Again, exception log is swallowed.